lxd/storage_volumes_snapshot.go

```diff
@@ -640,6 +673,25 @@ func autoCreateCustomVolumeSnapshotsTask(d *Daemon) (task.Func, task.Schedule) {
 				continue
 			}
 
+			// If there is more than one node (clustering), a stable random node is chosen to perform the snapshot.
+			if len(nodes) > 1 {
+				seed := fmt.Sprintf("%d", v.ID)
+
+				hash := fnv.New64a()
+				_, err = io.WriteString(hash, seed)
+				if err != nil {
+					continue
+				}
+
+				r := rand.New(rand.NewSource(int64(hash.Sum64())))
+				selectedNodeID := availableNodeIDs[r.Int63n(int64(len(availableNodeIDs)))]
```

Comment:
@monstermunchkin shouldnt this be `len(availableNodeIDs)-1` to avoid potential slice out of bound crashes?

Reply:
Shouldn't be necessary as `r.Int63n` will return a number between 0 and `len(availableNodes) - 1`.

Reply:
@monstermunchkin ah ok, the manual just says `[0,n` which is some strange notation I don't understand. Does it never return the number in `n`?

Reply:
The manual says `[0,n)` which is the same as `[0,n[`, so excluding `n`.

Reply:
@monstermunchkin ive not come across that notation before, have you got any resources on that, whats it called? Thanks

Reply:
Had to look it up earlier on Wikipedia myself: https://en.m.wikipedia.org/wiki/Interval_(mathematics)

Reply:
Cheers!